server/src/test/java/org/elasticsearch/cluster/InternalClusterInfoServiceSchedulingTests.java

```diff
@@ -105,6 +105,7 @@ protected PrioritizedEsThreadPoolExecutor createThreadPoolExecutor() {
             setFlagOnSuccess(becameMaster2)
         );
         runUntilFlag(deterministicTaskQueue, becameMaster2);
+        deterministicTaskQueue.runAllRunnableTasks();
```

Comment:
We now sometimes had a runnable (but failing on a block) task for the internal cluster info service ready to execute here that would mess up the count assertions in the loop below, so draining the executor here.